core/pl/stream.pm.sdoc

```diff
@@ -59,6 +59,7 @@ sub rw(&) {my @r = ($_); push @r, $_ while  defined rl && &{$_[0]}; push @q, $_
 sub ru(&) {my @r = ($_); push @r, $_ until !defined rl || &{$_[0]}; push @q, $_ if defined $_; @r}
 sub re(&) {my ($f, $i) = ($_[0], &{$_[0]}); rw {&$f eq $i}}
 BEGIN {ceval sprintf 'sub re%s() {re {%s}}', $_, $_ for 'a'..'l'}
+sub rn($) {my $max_val = shift; my @r = ($_); push @r, $_ while defined rl && (@r < $max_val); push @q, $_ if defined $_; @r} 
```

Comment:
actually, in this vein, the interaction of `a()` with `rea` is a little bit annoying, because once you `rea`, `a()` now points to the first line with the _new_ value of `a()`. I feel like I should be able to write `p'r like `@vals = uniq b_ rea; r a, @vals` where `a` would refer to the value of `a` over which the readahead took place.

e.g. ```$ni i[a 1] i[a 2] i[b 3] i[c 4] p'r F_ for(1..10)' p'$a = a; @b = b_ rea; r $a, a, uniq @b, $#b'
a       b       1       2       19
b       c       3       9
c               4       9
```